src/topology/sheaves/forget.lean

```diff
@@ -0,0 +1,209 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import topology.sheaves.sheaf
+import category_theory.limits.preserves.shapes
+import category_theory.limits.types
+
+/-!
+# Checking the sheaf condition on the underlying presheaf of types.
+
+If `G : C ⥤ D` is a functor which reflects isomorphisms and preserves limits
+(we assume all limits exist in both `C` and `D`),
+then checking the sheaf condition for a presheaf `F : presheaf C X`
+is equivalent to checking the sheaf condition for `F ⋙ G`.
+
+The important special case is when
+`C` is a concrete category with a forgetful functor
+that preserves limits and reflects isomorphisms.
+Then to check the sheaf condition it suffices
+to check it on the underlying sheaf of types.
+
+## References
+* https://stacks.math.columbia.edu/tag/0073
+-/
+
+open category_theory
+open category_theory.limits
+open topological_space
+open opposite
+
+
+namespace Top
+
+namespace sheaf_condition
+
+universes v u₁ u₂
+
+variables {C : Type u₁} [category.{v} C] [has_limits C]
+variables {D : Type u₂} [category.{v} D] [has_limits D]
+variables (G : C ⥤ D) [preserves_limits G]
+variables {X : Top.{v}} (F : presheaf C X)
+variables {ι : Type v} (U : ι → opens X)
+
+local attribute [reducible] sheaf_condition.diagram
+local attribute [reducible] sheaf_condition.left_res sheaf_condition.right_res
+
+/--
+When `G` preserves limits, the sheaf condition diagram for `F` composed with `G` is
+naturally isomorphic to the sheaf condition diagram for `F ⋙ G`.
+-/
+def diagram_comp_preserves_limits :
+  sheaf_condition.diagram F U ⋙ G ≅ sheaf_condition.diagram (F ⋙ G) U :=
+begin
+  fapply nat_iso.of_components,
+  intro j,
```

Comment:
Do you want to golf these `intro, cases` into `rintro` statements? Also line 61.
Or would you prefer to keep them like this?

Reply:
Done!